chain/client/src/client.rs

```diff
@@ -2257,11 +2264,39 @@ impl Client {
                     .insert(bp.public_key().clone());
             }
         }
+        for v in self.epoch_manager.get_epoch_all_validators(&tip.epoch_id)? {
+            account_keys.entry(v.account_id().clone()).or_default().insert(v.public_key().clone());
+        }
         let account_keys = Arc::new(account_keys);
         self.tier1_accounts_cache = Some((tip.epoch_id, account_keys.clone()));
         Ok(account_keys)
     }
 
+    /// We send the optimistic block to the chunk producers of the current and next epochs.
+    pub(crate) fn get_optimistic_block_targets(
+        &mut self,
+        tip: &Tip,
+    ) -> Result<Arc<Vec<AccountId>>, Error> {
+        match &self.optimistic_block_targets_cache {
+            Some(it) if it.0 == tip.epoch_id => return Ok(it.1.clone()),
```

Comment:
Can you get it by reference instead of cloning? 
nit: Can you unwrap it further?

Reply:
Ah I see it's an arc so clone should be cheap. Perhaps make it explicit with Arc::clone to make it less scary (guessing this syntax but there should be something like that).